### ISSUE TITLE
Respect XDG Base Directory Specification on Linux

### ISSUE THREAD

**Issue by LiteracyFanatic** [REditorSupport/vscode-R#495]
**Is your feature request related to a problem? Please describe.**
The [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) is a widely followed set of guidelines for where programs should store their files on Linux. Besides keeping the user's home directory free from clutter, the XDG specification helps to simplify user configuration, backup, and source control scenarios. Would you please consider changing the location of the files in `~/.vscode-R/` to follow the XDG specification? 

**Describe the solution you'd like**
Basically, any files intended to be edited by the user should go in `$XDG_CONFIG_HOME/vscode-R/`, any that store application state or data that can't be recreated if lost (i.e. log files) should go in `$XDG_DATA_HOME/vscode-R/`, and any files that can be automatically recreated belong in `$XDG_CACHE_HOME/vscode-R/`. If the environment variables are undefined, you should use the following default values:

```
XDG_CONFIG_HOME="$HOME/.config"
XDG_DATA_HOME="$HOME/.local/share"
XDG_CACHE_HOME="$HOME/.cache"
```
